File: train.js

// TASK - F

// ❓ Shunday findDoublers function tuzing, unga faqat bitta string argument pass bolib,
// agar stringda bir hil harf qatnashgan bolsa true,
// qatnashmasa false qaytarishi kerak.

// ✅ Masalaning yechimi:
function findDoublers(str) {
    let counter = {};
    for (let harf of str) {
        if (counter[harf]) return true;
        counter[harf] = 1;
    }
    return false;
}

console.log(findDoublers("hello"));
console.log(findDoublers("world"));






// TASK - E

// ❓ Shunday function tuzing, u bitta string argumentni qabul qilib 
// osha stringni teskari qilib return qilsin.


// ✅ Masalaning yechimi:
// function getReverse(str) {
//     let result = "";
//     for (let i = str.length - 1; i >= 0; i--) {
//         result += str[i];
//     }
//     return result;
// }

// console.log(getReverse("NodeJS"));

// TASK - D

// ❓ Shunday function tuzingki unga integerlardan iborat array pass bolsin
// va function bizga osha arrayning eng katta qiymatiga tegishli 
// birinchi indexni qaytarsin.

// // ✅ Masalaning yechimi:
// function getArray(arr) {
//     let katta = arr[0]
//     let kattaIndex = 0;
//     for (let i = 0; i < arr.length; i++) {
//         if (arr[i] > katta) {
//             katta = arr[i]
//             kattaIndex = i
//         }
//     }

//     return kattaIndex
// }

// console.log(getArray([5, 11, 7, 24, 18]));



// TASK-C

// ❓ Shunday function tuzing, u 2ta string parametr ega bolsin,
//hamda agar har ikkala string bir hil harflardan iborat bolsa true
//aks holda false qaytarsin

// // ✅ Masalaning yechimi:
// function sameLetters(a, b) {
//     let sort1 = a.split("").sort().join("");
//     let sort2 = b.split("").sort().join("");


//     return sort1 === sort2;


// }
// result = sameLetters("night", "thing");
// console.log(result);


// TASK-B

// ❓ Shunday function tuzing, u 1ta string parametrga ega bolsin,
// hamda osha stringda
// qatnashgan raqamlarni sonini bizga return qilsin.

// ✅ Masalaning yechimi:

// function countNumbers(str) {
//     let count = 0;

//     for (let char of str) {
//         if (char >= '0' && char <= '9') {
//             count++;
//         }
//     }

//     return count;
// }

// let result = countNumbers("ajd212efdqfq");
// console.log("result:", result);


// /* TASK-A
// ❓ Shunday 2 parametrli function tuzing, hamda birinchi parametrdagi letterni ikkinchi
// parametrdagi sozdan qatnashga sonini return qilishi kerak boladi.
// */

// // ✅ Masalaning yechimi:
// function mevaHarfi(a, b) {
//     let soni = 0;

//     for (let harf of b) {
//         if (harf === a) {
//             soni++
//         }
//     }

//     return soni;
// }

// let result = mevaHarfi("o", "avocado");
// console.log(result)

File: train.py
# M - TASK

# ❓ Shunday function yozing, u string qabul qilsin va string palindrom
# yani togri oqilganda ham, orqasidan oqilganda ham bir hil oqiladigan soz
# ekanligini aniqlab boolean qiymat qaytarsin.

# ✅ Masalaning yechimi:
def palindrom_check(s: str) -> bool:
    return s == s[::-1]


print(palindrom_check("level"))


# K-TASK

# ❓ Shunday function yozing, u string qabul qilsin
# va string ichidagi eng uzun sozni qaytarsin.
# MASALAN: find_longest("I come from Uzbekistan") return "Uzbekistan"

# # ✅ Masalaning yechimi:
# def find_longest(text):
#     words = text.split()
#     return max(words, key=len)


# print(find_longest("Python dasturlash tili"))

# I-TASK

# ❓ Shunday function tuzing, unga string argument pass bolsin.
# Function ushbu agrumentdagi
# digitlarni yangi stringda return qilsin

# ✅ Masalaning yechimi:
# def get_digits(string):
#     result = ""

#     for char in string:
#         if char.isdigit():
#             result += char

#     return result


# print(get_digits("m14i1t"))


# G-TASK

# ❓ Shunday function tuzingki unga integerlardan iborat array pass bolsin
# va function bizga osha arrayning eng katta qiymatiga tegishli
# birinchi indexni qaytarsin.

# MASALAN: get_highest_index([5, 21, 12, 21, 8]) return qiladi 1 sonini.


# ✅ Masalaning yechimi:
# def sonlar(list):
#     eng_kata = 0

#     for s in range(0, len(list)):
#         if list[s] > list[eng_kata]:
#             eng_kata = s

#     return eng_kata


# print(sonlar([12, 6, 13, 34, 16]))

# ❓ Shunday funksiya yozingki, unga integerlardan iborat array
# berilsin va funksiya arraydagi
# eng kichik sonning birinchi indeksini qaytarsin.


# def eng_kichik(son):
#     kichik = 0

#     for x in range(0, len(son)):
#         if son[x] < son[kichik]:
#             kichik = x

#     return kichik


# print(eng_kichik([15, 9, 1, 12, 7]))


# ❓ Shunday funksiya yozingki, unga integerlardan iborat
# array berilsin va funksiya arraydagi barcha
# sonlarning yig'indisini qaytarsin.
# def sum_ofall(sonlar):
#     boshla = 0

#     for x in range(0, len(sonlar)):
#         boshla = boshla + sonlar[x]

#     return boshla


# print(sum_ofall([1, 5, 7, 9, 3]))
